testsuite/shell.go

```diff
@@ -128,103 +85,57 @@ func (shell *ShellInstance) ConfigureTypeOfShell(shellName string) {
 		switch runtime.GOOS {
 		case "darwin", "linux":
 			shell.name = "bash"
-			shell.checkExitCodeCmd = fmt.Sprintf(bashExitCodeCheck, exitCodeIdentifier)
 		case "windows":
 			shell.name = "powershell"
 			shell.startArgument = []string{"-Command", "-"}
-			shell.checkExitCodeCmd = fmt.Sprintf(powershellExitCodeCheck, exitCodeIdentifier)
 		}
 	}
-
 	return
 }
 
-func StartHostShellInstance(shellName string) error {
-	return shell.Start(shellName)
+func SetUpHostShellInstance(shellName string) error {
+	return shell.SetUp(shellName)
 }
 
-func (shell *ShellInstance) Start(shellName string) error {
-	var err error
-
+func (shell *ShellInstance) SetUp(shellName string) error {
 	if shell.name == "" {
 		shell.ConfigureTypeOfShell(shellName)
 	}
-	shell.exitCodeChannel = make(chan string)
-
-	shell.instance = exec.Command(shell.name, shell.startArgument...)
-
-	shell.outPipe, err = shell.instance.StdoutPipe()
-	if err != nil {
-		return err
-	}
-
-	shell.errPipe, err = shell.instance.StderrPipe()
-	if err != nil {
-		return err
-	}
-
-	shell.inPipe, err = shell.instance.StdinPipe()
-	if err != nil {
-		return err
-	}
-
-	shell.outScanner = bufio.NewScanner(shell.outPipe)
-	shell.errScanner = bufio.NewScanner(shell.errPipe)
-
-	go shell.ScanPipe(shell.outScanner, &shell.outbuf, "stdout")
-	go shell.ScanPipe(shell.errScanner, &shell.errbuf, "stderr")
-
-	err = shell.instance.Start()
-	if err != nil {
-		return err
-	}
 
 	fmt.Printf("The %v instance has been started and will be used for testing.\n", shell.name)
-	return err
-}
-
-func CloseHostShellInstance() error {
-	return shell.Close()
-}
-
-func (shell *ShellInstance) Close() error {
-	closingCmd := "exit\n"
-	io.WriteString(shell.inPipe, closingCmd)
-	err := shell.instance.Wait()
-	if err != nil {
-		fmt.Println("error closing shell instance:", err)
-	}
-
-	shell.instance = nil
-
-	return err
+	return nil
 }
 
 func ExecuteCommand(command string) error {
-	if shell.instance == nil {
+	if shell == nil {
 		return errors.New("shell instance is not started")
 	}
-
+	var (
+		exitError *exec.ExitError
+		pathError *os.PathError
+	)
+	shell.exitCode = 0
 	shell.outbuf.Reset()
 	shell.errbuf.Reset()
-	shell.excbuf.Reset()
 
 	util.LogMessage(shell.name, command)
 
-	_, err := io.WriteString(shell.inPipe, command+"\n")
-	if err != nil {
-		return err
-	}
-
-	_, err = shell.inPipe.Write([]byte(shell.checkExitCodeCmd + "\n"))
-	if err != nil {
-		return err
+	cmd := exec.Command(shell.name, shell.startArgument...)
+	cmd.Stderr = &shell.errbuf
+	cmd.Stdout = &shell.outbuf
+	cmd.Stdin = strings.NewReader(command + "\n")
```

Comment:
The log says
> We still create the input Pipe and pass the command in a goroutine but just after that
executing this stdinPipe closed 

I think this is no longer  correct?